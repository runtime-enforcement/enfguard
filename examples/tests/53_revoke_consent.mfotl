ALWAYS (FORALL x. use(x) IMPLIES ((NOT revoke(x)) SINCE consent(x)))