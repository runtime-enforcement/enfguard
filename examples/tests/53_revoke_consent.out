The formula
 □[0s,∞) (∀x:int. use(x:int) → ¬revoke(x:int) S[0s,∞) consent(x:int))
is enforceable and types to
 □[0s,∞) ((∀x:int. ((use(x:int) : Sup) →:L ¬revoke(x:int) S[0s,∞) consent(x:int) : Cau)) : Cau) : Cau
[Enforcer] @1 OK.
[Enforcer] @1 nothing to do proactively.
[Enforcer] @2 OK.
[Enforcer] @2 nothing to do proactively.
[Enforcer] @3 OK.
[Enforcer] @3 nothing to do proactively.
[Enforcer] @4 reactively commands:
Suppress:
use(2)
OK.
[Enforcer] @4 nothing to do proactively.
[Enforcer] @5 reactively commands:
Suppress:
use(3)
OK.
